client/components/site-selector/index.jsx

```diff
@@ -480,7 +480,16 @@ const navigateToSite =
 				const path = allSitesPath.replace( /^\/posts\b(\/my)?/, postsBase );
 
 				// There is currently no "all sites" version of the insights page
```

Comment:
Thanks, I removed the slash from the regex, so it brought the current behavior back.

I'm not sure if I understand the change for `/stats/insights` you proposed. Currently, there is no "all sites" version for the insights page, so this page shows the site selector. Do you mean to investigate if we should remove this condition completely, to serve the site selector in this case instead of redirecting the user to the "all sites" version of `/stats/day`?

Reply:
Sorry, maybe I misunderstood. Isn't the "all sites version" the site selector?

>Do you mean to investigate if we should remove this condition completely, to serve the site selector in this case instead of redirecting the user to the "all sites" version of /stats/day?

Yes 👍

Reply:
@zaguiini, when this docblock comment says "all sites version", it means the version of the page that displays aggregated entities for all sites owned by the user. Example pages are `/posts/my` or `/stats/day`. Thus if you are on posts or stats pages for a single site, and you use the "All My Sites" link, it won't show you the site selector but will show the "all sites version" for a given entity.

If a given page doesn't have an "all sites version", the "All My Sites" link leads to the site selector. The example pages are `/media` or `/comments/all`.

In the condition we are fixing here, the author wanted to explicitly make `/stats/insights` redirect to `/stats/day` instead of displaying the site selector. I tracked it to [the initial wp-calypso commit](https://github.com/Automattic/wp-calypso/blame/51a8e1f4cdb5d06e6a71b2ac5cba557edbcbdb56/client/components/site-selector/index.jsx#L216 ), I couldn't find an older changeset that could explain why we did it that way.

Reply:
@zaguiini  I've added https://github.com/Automattic/wp-calypso/issues/65723 to change this (or to review if this is expected).